x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Page.java

```diff
@@ -131,6 +131,9 @@ public <B extends Block> B getBlock(int blockIndex) {
         if (blocksReleased) {
             throw new IllegalStateException("can't read released page");
         }
+        if (blockIndex < 0) {
+            blockIndex += blocks.length;
+        }
```

Comment:
I think I'd prefer we did this in `PackegeValueBlockHash` if we're going to use the negative index.